CMakeLists.txt

```diff
@@ -84,8 +110,31 @@ if(IS_LOOKUP_NODE)
         SET( CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${GCC_COVERAGE_LINK_FLAGS} -pthread -ljsoncpp -lboost_system -lboost_filesystem -std=c++14 -Wall -DSTAT_TEST -DIS_LOOKUP_NODE" )
     endif()
     if(${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
-        SET( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${GCC_COVERAGE_COMPILE_FLAGS} -pthread -stdlib=libc++ -std=c++14 -Wall -DSTAT_TEST -DIS_LOOKUP_NODE -ggdb -I/usr/local/Cellar/openssl/1.0.2/include/openssl/ -I/usr/local/Cellar/boost/1.64.0_1/include -I/usr/local/include" )
-        SET( CMAKE_EXE_LINKER_FLAGS "-L/usr/local/opt/boost@1.64/lib ${CMAKE_EXE_LINKER_FLAGS} ${GCC_COVERAGE_LINK_FLAGS} -pthread ${jsoncpp_LIBRARIES}  -lboost_system -lboost_filesystem -std=c++14 -Wall -DSTAT_TEST -DIS_LOOKUP_NODE -L/usr/local/Cellar/openssl/1.0.2/lib" )
+        SET( CMAKE_CXX_FLAGS
+            "${CMAKE_CXX_FLAGS} \
+            ${GCC_COVERAGE_COMPILE_FLAGS} \
+            -pthread \
```

Comment:
:bulb:

```
add_compile_options(-pthread)
add_compile_options(-stdlib=libc++)
add_compile_options(-std=c++14)
add_compile_options(-Wall)
```

Reply:
`add_compile_options` adds flags for both C and C++. `add_compile_options(-std=c++14)` actually cause an C compiler warning of unrecognized flags (we have C compling in some directories). A better solution is still using `set(CMAKE_CXX_FLAGS ...)` for these.